ui/apps/platform/src/services/ReportsService.ts

```diff
@@ -119,8 +119,36 @@ export function fetchReportHistory({
         });
 }
 
+export type FetchOnDemandReportHistoryServiceProps = {
+    query: string;
+    page: number;
+    perPage: number;
+    sortOption: ApiSortOption;
+    showMyHistory: boolean;
+};
+
 // @TODO: Pass API query information and set up API call to endpoint
-export function fetchOnDemandReportHistory(): Promise<OnDemandReportSnapshot[]> {
+export function fetchOnDemandReportHistory({
+    query,
+    page,
+    perPage,
+    sortOption,
+    // @TODO: Use the showMyHistory value to determine which endpoint to use
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    showMyHistory,
+}: FetchOnDemandReportHistoryServiceProps): Promise<OnDemandReportSnapshot[]> {
+    // @TODO: Use the params in the future API call
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const params = queryString.stringify(
+        {
+            reportParamQuery: {
+                query,
+                pagination: getPaginationParams({ page, perPage, sortOption }),
+            },
+        },
+        { arrayFormat: 'repeat', allowDots: true }
+    );
```

Comment:
Two thoughts:
1. Is the future API going to wrap the `query` and `pagination` parameters under a top level `reportParamQuery`? If so, could this be avoided? It would add an unfortunate extra layer of inconsistency with many of our other APIs.
2. Could we utilize something like `searchUtils.ts#getListQueryParams()` to encapsulate this logic? It looks like some code moves in that direction, although there are a lot of places that duplicate the `stringify` calls throughout our code as well.

Reply:
Since the Backend is using the same existing structures in place for the already defined reporting-related APIs, I don't think we'll be able to avoid it 😢. During the planning phase, we would have considered improving things. I'll double-check with Backend, though.

Reply:
We can utilize `getListQueryParams`. It looks like it works well with what I have. I'll make that change.